supplier/tests/test_supplier.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+from conftest import Fields, TestFields
+from django.core.exceptions import FieldDoesNotExist, ValidationError
+from supplier.models import Supplier
+
+
+class TestSupplierModel:
+    @pytest.mark.django_db()
+    def test_save_supplier(self, supplier0):
+        """Test - checks if supplier is saved in DB.
+
+        Args:
+            supplier0: fixture from module conftest.py.
+
+        Returns:
+            None.
+        """
+        Supplier.save_supplier(supplier0)
+        assert supplier0 in Supplier.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_supplier(self, saved_supplier0):
+        """Test - checks if supplier is not in DB after deletion.
+
+        Args:
+            saved_supplier0: fixture from module conftest.py.
+
+        Returns:
+            None.
+        """
+        Supplier.delete_supplier(saved_supplier0)
+        assert saved_supplier0 not in Supplier.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_field(self, saved_supplier0):
+        """Test - checks the Supplier method filter by field.
+
+        The test checks whether the fixtured Supplier - saved_supplier0 -
+        can be found in the returned query set of each attribute and
+        attribute value given to the method "filter by field".
+
+        Args:
+            saved_supplier0: fixture from module conftest.py.
+
+        Returns:
+            None.
+        """
+        for attri, testAttri in zip(Fields, TestFields):
+            assert saved_supplier0 in Supplier.filter_by_field(attri.name,
```

Comment:
How do you know it does not return all products? i.e. need to also assert that there could be queries that does not find what you're looking for.

Reply:
```
   @staticmethod
    def filter_by_field(field, field_value):
        """Fetches list of Suppliers filtered with given field and given value.
        Args:
        field:
            Supplier attribute.
        field_value:
            value which will filter the suppliers.
        Returns:
            returns a queryset containing the objects that match the specified filter conditions.
        Raises:
        FieldDoesNotExist error: if Supplier doesn't have the specified attribute.
        """
        if hasattr(Supplier, str(field)):
            return Supplier.objects.filter(**{field: field_value})
        raise FieldDoesNotExist
```

it could return all the *Suppliers*, also raises an error.

Reply:
I'm saying that you need a test in which this query does not find what you're looking for.